src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -400,6 +402,28 @@ public virtual async Task<ConnectionCompleteParams> Connect(ConnectParams connec
             return completeParams;
         }
 
+        private async Task<ConnectionCompleteParams> getResponseConnect(ConnectionInfo connectionInfo, ConnectParams connectionParams)
+        {
+            Boolean firstRun = true;
+            int counter = 0;
+            ConnectionCompleteParams response = null;
+            while (firstRun && counter <= MaxTries)
+            {
+                response = await TryOpenConnection(connectionInfo, connectionParams);

```

Comment:
So TryOpenConnection will never throw, right? if not you have to wrap this with try catch block.

Reply:
It already catches for errors in the areas that it is expected in, so adding another Try Catch block is redundant.